ksp_plugin_adapter/ksp_plugin_adapter.cs

```diff
@@ -268,9 +268,15 @@ where is_in_physics_bubble(vessel)
     base.OnSave(node);
     if (PluginRunning()) {
       IntPtr serialization = IntPtr.Zero;
+      IntPtr serializer = IntPtr.Zero;
       try {
-        serialization = SerializePlugin(plugin_);
-        node.AddValue(kPrincipiaKey, Marshal.PtrToStringAnsi(serialization));
+        for (;;) {
+          serialization = SerializePlugin(plugin_, ref serializer);
```

Comment:
The previous serialization hasn't been deleted, you're leaking memory like a broken snowden.